packages/amplify_auth_plugin_interface/lib/src/SignIn/SignInWithWebUIOptions.dart

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+abstract class SignInWithWebUIOptions {
+  const SignInWithWebUIOptions();
+  Map<String, dynamic> serializeAsMap() {
+    throw UnimplementedError(
+        'serializeAsMap() has not been implemented on SignInWithWebUIOptions.');
+  }
```

Comment:
Prefer abstract method

```suggestion
  Map<String, dynamic> serializeAsMap();
```

Reply:
Thank you! I have updated the code